src/include/duckdb/execution/aggregate_hashtable.hpp

```diff
@@ -139,24 +141,23 @@ class GroupedAggregateHashTable : public BaseAggregateHashTable {
 	idx_t MaxCapacity();
 	static idx_t GetMaxCapacity(HtEntryType entry_type, idx_t tuple_size);
 
-	void Partition(vector<GroupedAggregateHashTable *> &partition_hts, hash_t mask, idx_t shift);
+	void Partition(vector<GroupedAggregateHashTable *> &partition_hts, idx_t radix_bits);
+	void InitializeFirstPart();
 
 	void Finalize();
 
 private:
 	HtEntryType entry_type;
 
-	//! The total tuple size
+	//! The capacity of the HT. This can be increased using GroupedAggregateHashTable::Resize
+	idx_t capacity;
+	//! Tuple width
 	idx_t tuple_size;
-	//! The amount of tuples that fit in a single block
+	//! Tuples per block
 	idx_t tuples_per_block;
-	//! The capacity of the HT. This can be increased using
-	//! GroupedAggregateHashTable::Resize
-	idx_t capacity;
-	//! The amount of entries stored in the HT currently
-	idx_t entries;
 	//! The data of the HT
-	vector<BufferHandle> payload_hds;
+	unique_ptr<TupleDataCollection> data_collection;
+	TupleDataAppendState td_append_state;
```

Comment:
Should this be moved to the `AggregateHTAppendState`?

Reply:
Good point, I will try to do this. I can move the `TupleDataChunkState` of the `TupleDataAppendState` into the `AggregateHTAppendState`, but the `TupleDataPinState` of the `TupleDataAppendState` needs to be HT-specific.